src/logging.c

```diff
@@ -414,6 +414,92 @@ static const char * copy_from_orig(
   return logging_output->buffer;
 }
 
+// copy buffers and decode the escape characters if they're exists
+static const char * decode_orig(
+  const logging_input_t * logging_input,
+  rcutils_char_array_t * logging_output,
+  size_t start_offset, size_t end_offset)
+{
+  size_t back_slash_index = rcutils_findn(
+    g_rcutils_logging_output_format_string + start_offset, '\\', end_offset - start_offset);
+  if (back_slash_index == SIZE_MAX) {
+    // back slash not found, copy full buffers
+    return copy_from_orig(logging_input, logging_output, start_offset, end_offset);
+  } else {
+    // TODO(anybody): optimize the following if possible
+    size_t end_index = back_slash_index;
+    for (size_t i = start_offset + end_index; i < end_offset; ++i) {
+      if (g_rcutils_logging_output_format_string[i] == '\\') {
+        // found back slash, copy previous buffer if it's available
+        if (end_index != 0) {
+          if (rcutils_char_array_strncat(
+              logging_output,
+              g_rcutils_logging_output_format_string + start_offset,
+              end_index) != RCUTILS_RET_OK)
+          {
+            RCUTILS_SAFE_FWRITE_TO_STDERR(rcutils_get_error_string().str);
+            rcutils_reset_error();
+            RCUTILS_SAFE_FWRITE_TO_STDERR("\n");
+            return NULL;
+          }
+          end_index = 0;
+        }
+
+        if (i + 1 < end_offset) {
+          const char * expected_char = NULL;
+          switch (g_rcutils_logging_output_format_string[i + 1]) {
+            case 'a':  expected_char = "\a"; break;
+            case 'b':  expected_char = "\b"; break;
+            case 'f':  expected_char = "\f"; break;
+            case 'n':  expected_char = "\n"; break;
+            case 'r':  expected_char = "\r"; break;
+            case 't':  expected_char = "\t"; break;
+            case 'v':  expected_char = "\v"; break;
```

Comment:
I am not sure if we need to support these, this needs to be discussed. besides, i would suggest the doc for each escape sequence.

```suggestion
            case 'a':  expected_char = "\a"; break; // alert
            case 'b':  expected_char = "\b"; break; // backspace
            case 'f':  expected_char = "\f"; break; // form feed
            case 'n':  expected_char = "\n"; break; // new line
            case 'r':  expected_char = "\r"; break; // carriage return
            case 't':  expected_char = "\t"; break; // horizontal tab
            case 'v':  expected_char = "\v"; break; // vertical tab
```

Reply:
Updated in https://github.com/ros2/rcutils/pull/426/commits/097c219af9f5b382614aa270d316f278d5051a21